### ISSUE TITLE
Way to merge when conflicts exist

### ISSUE THREAD

**Issue by saschanaz** [octokit/rest.js#1626]
I want to push a merge commit where the conflicting files are rewritten by the tool. 

Background:

I have a tool named [`webidl-updater`](https://github.com/saschanaz/webidl-updater/) to read Web IDL from web specifications and send pull requests to modify their syntax problems automatically.

The PRs frequently get merge conflicts since the specs are being actively updated, but there is no way to solve the conflicts solely with Octokit REST API.

Currently the tool just opens a new PR whenever conflict happens, which is suboptimal. (See https://github.com/w3c/csswg-drafts/pull/4385 and https://github.com/w3c/csswg-drafts/pull/4798.)

**Comment by gr2m** [comment]
Hi @saschanaz, very cool to see that you use Octokit for the CSS Working Group drafts!

Just so I understand it right: you want your bot account to create merge commits to resolve conflicts in current pull requests? In the [`forceUpdateToLatestCommit()` method](https://github.com/saschanaz/webidl-updater/blob/13cc6ea5dd714c03d5d0bb0cf3af383d832e24bc/src/github-pullrequest.js#L170-L178
)?

Is the use case to basically rebase open pull requests on latest master?

**Comment by saschanaz** [comment]
>Just so I understand it right

Right, but only for existing PRs that the bot opened, because the bot knows what should be done for them.

>Is the use case to basically rebase open pull requests on latest master?

That's what I do as a workaround for now, but rebasing always closes existing PRs and thus causes a need to open new PRs whenever conflicts happen.

**Comment by gr2m** [comment]
> but rebasing always closes existing PRs and thus causes a need to open new PRs whenever conflicts happen

@bcoe I think you run into this challenge before? Did you end up resolving it?

**Comment by bcoe** [comment]
👋 for our case, we always update the branch from `master` as we're creating new PRs, and I believe hav rarely run into conflicts (@chingor13 did manage to get into a conflicting state once, and I don't know that we got to the bottom of it, perhaps he can speak to this).

The problem we were running into, was making the PR not close and reopen every time we update from `HEAD`.

**Comment by saschanaz** [comment]
>was making the PR not close and reopen every time we update from `HEAD`.

You mean you solved that problem? I wonder how 🧐

**Comment by bcoe** [comment]
@saschanaz here's the logic that's mostly been working for us for maintaining branches:

https://github.com/googleapis/release-please/blob/master/src/github.ts#L569

**Comment by saschanaz** [comment]
It's still [marked as TODO](https://github.com/googleapis/release-please/blob/de3d557c49dbded349a630fa6f5aa8cd5a2a2112/src/github.ts#L660), so maybe you just haven't recently run into conflicts as you said...

**Comment by gr2m** [comment]
if you could create a script that reproduces the problem, I'd be happy to look into it myself, too. It sounds like a common-enough use case to create a reusable plugin for

**Comment by kevinbuhmann** [comment]
I have a similar use case. Here's my workaround adapted for this scenario:

1. Create a temp branch from `master`.
2. Create or update the file.
3. Force update the original PR branch with the commit sha from step 2.
4. Delete the temp branch.
